### PR TITLE
Split google k8s provider from provider

### PR DESCRIPTION
**Pull request by ilausuch**
- Related ticket: https://progress.opensuse.org/issues/101497
- Verification run: 
-- http://copland.arch.suse.de/tests/607 (GKE)
-- http://copland.arch.suse.de/tests/606 (EKS)
-- https://openqa.suse.de/tests/7762439 (GCE)

### CODE REVIEW

File: lib/publiccloud/gcp_client.pm [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+sub create_credentials_file {
+    my ($self) = @_;
+    my $credentials_file;
+
+    if ($self->private_key_id()) {
+        $credentials_file
+          = "{"
+          . $/
```
Comment (pdostal) [inline lib/publiccloud/gcp_client.pm:77]: What is this for?

File: lib/publiccloud/gcp_client.pm [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          . '"private_key_id": "'
+          . $self->private_key_id . '", '
+          . $/
+          . '"private_key": "'
+          . $self->private_key . '", '
+          . $/
+          . '"client_email": "'
+          . $self->service_acount_name . '@'
+          . $self->project_id
+          . '.iam.gserviceaccount.com", '
```
Comment (pdostal) [inline lib/publiccloud/gcp_client.pm:92]:
```suggestion
          . '"client_email": "' . $self->service_acount_name . '@' . $self->project_id . '.iam.gserviceaccount.com", '
```
Please make the `$credentials_file` more compact.

File: lib/publiccloud/gcp_client.pm (hunk 1 above)
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm:77]: Perltidy is doing this. :(

File: lib/publiccloud/gcp_client.pm (hunk 2 above)
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm:92]: Perltidy is doing this. :(

File: lib/publiccloud/gcp_client.pm (hunk 1 above)
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm:77]: not sure, was in the original code https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/dc7536355ee28f9b1669a5453977c010124cb164/lib/publiccloud/gcp.pm#L75

File: lib/publiccloud/gcp_client.pm (hunk 2 above)
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm]: Ok, tidy was saying that the work was not done. I created a new commit use the gcp_client logic directly on the gce library

**Review by pdostal — APPROVED** [review]

File: lib/publiccloud/aws_client.pm [hunk 3]
```diff
@@ -120,6 +120,17 @@ sub get_container_image_full_name {
     return "$full_name_prefix/" . $self->container_registry . ":$tag";
 }
 
+=head2 configure_docker
+Configure the docker to access the cloud provider registry
+=cut
```
Comment (b10n1k) [inline lib/publiccloud/aws_client.pm:125]: empty line

File: lib/publiccloud/aws_client.pm [hunk 4]
```diff
@@ -120,6 +120,17 @@ sub get_container_image_full_name {
     return "$full_name_prefix/" . $self->container_registry . ":$tag";
 }
 
+=head2 configure_docker
+Configure the docker to access the cloud provider registry
```
Comment (b10n1k) [inline lib/publiccloud/aws_client.pm:124]: empty line

File: lib/publiccloud/gcp_client.pm [hunk 5]
```diff
@@ -27,27 +25,28 @@ has service_acount_name => undef;
...
-        get_var('PUBLIC_CLOUD_VAULT_ROLES', 'openqa-role,openqa-role1,openqa-role2,openqa-role3'));
+    return split(/\s*,\s*/, get_var('PUBLIC_CLOUD_VAULT_ROLES', 'openqa-role,openqa-role1,openqa-role2,openqa-role3'));
 }
 
-
 sub init {
-    my ($self) = @_;
-    $self->SUPER::init();
-    $self->vault(publiccloud::vault->new());
+    my ($self, %params) = @_;
```
Comment (b10n1k) [inline lib/publiccloud/gcp_client.pm:35]: `%params` are not used. am i right?

**Review by b10n1k — CHANGES_REQUESTED** [review]
about the perlpod.
As it is now returns (example)
```
 install_kubectl
Install kubectl from the k8s page
  apply_manifest
Apply a kubernetes manifest
```
when with the recommended changes you get (note only the `install_kubectl` how it looks compared to `apply_manifest`)
```
  install_kubectl
    Install kubectl from the k8s page

  apply_manifest
Apply a kubernetes manifest
```

File: lib/publiccloud/gcp_client.pm [hunk 6]
```diff
@@ -27,27 +25,28 @@ has service_acount_name => undef;
 has client_id => undef;
 has vault_gcp_role_index => undef;
 has gcr_zone => undef;
+has vault => undef;
 
 sub vault_gcp_roles {
-    return split(/\s*,\s*/,
-        get_var('PUBLIC_CLOUD_VAULT_ROLES', 'openqa-role,openqa-role1,openqa-role2,openqa-role3'));
+    return split(/\s*,\s*/, get_var('PUBLIC_CLOUD_VAULT_ROLES', 'openqa-role,openqa-role1,openqa-role2,openqa-role3'));
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm:31]: This code is inhered from an older code, so it's already tested. I am not sure if this change should be for a new PR

File: lib/publiccloud/gcp_client.pm (hunk 5 above)
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm]: done

File: lib/publiccloud/gcp_client.pm (hunk 6 above)
[the remark this answers is not in the document]
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm]: done, testing

File: lib/publiccloud/aws_client.pm (hunk 4 above)
Comment (ilausuch) [reply lib/publiccloud/aws_client.pm:124]: done, thanks

File: lib/publiccloud/aws_client.pm (hunk 3 above)
Comment (ilausuch) [reply lib/publiccloud/aws_client.pm:125]: done

File: lib/publiccloud/aws_client.pm (hunk 4 above)
Comment (b10n1k) [reply lib/publiccloud/aws_client.pm:124]: you missed this file

File: lib/publiccloud/gcp_client.pm (hunk 6 above)
[the remark this answers is not in the document]
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm]: Worked

[the remark this answers is not in the document]
Comment (ilausuch) [reply lib/publiccloud/gcp_client.pm]: Any concerns?

File: lib/publiccloud/aws_client.pm (hunk 4 above)
Comment (ilausuch) [reply lib/publiccloud/aws_client.pm:129]: yes, thank you done

File: lib/publiccloud/gcp_client.pm (hunk 6 above)
[the remark this answers is not in the document]
Comment (cfconrad) [reply lib/publiccloud/gcp_client.pm]: I would keep it as it is. The `\s*` are only sanity, if someone make some spaces arround the `,`.

**Review by b10n1k — APPROVED** [review]
LGTM

**Review by jlausuch — APPROVED** [review]
LGTM

File: lib/publiccloud/gcp_client.pm [hunk 7]
```diff
@@ -27,35 +25,36 @@ has service_acount_name => undef;
 has client_id => undef;
 has vault_gcp_role_index => undef;
 has gcr_zone => undef;
-
-sub vault_gcp_roles {
-    return split(/\s*,\s*/,
-        get_var('PUBLIC_CLOUD_VAULT_ROLES', 'openqa-role,openqa-role1,openqa-role2,openqa-role3'));
-}
-
+has vault => undef;
```
Comment (cfconrad) [inline lib/publiccloud/gcp_client.pm:28]:
```
has vault => sub { publiccloud::vault->new() };
```

File: lib/publiccloud/gcp_client.pm [hunk 8]
```diff
@@ -27,35 +25,36 @@ has service_acount_name => undef;
...
 
 sub init {
     my ($self) = @_;
-    $self->SUPER::init();
-    $self->vault(publiccloud::vault->new());
 
     $self->gcr_zone(get_var('PUBLIC_CLOUD_GCR_ZONE', 'eu.gcr.io'));
 
+    $self->vault(publiccloud::vault->new());
```
Comment (cfconrad) [inline lib/publiccloud/gcp_client.pm:35]: see `has vault`

**Pull request merged**